src/Symfony/Component/HttpFoundation/Exception/AbstractHostException.php

```diff
@@ -0,0 +1,43 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\HttpFoundation\Exception;
+
+/**
+ * THe HTTP request contains incorrect host data.
+ *
+ * @author SpacePossum
+ */
+abstract class AbstractHostException extends \UnexpectedValueException implements ExceptionInterface
+{
+    /**
+     * @var string
+     */
+    private $host;
+
+    /**
+     * @param string $host
+     * @param string $message
+     */
+    public function __construct($host, $message)
```

Comment:
Making it `null` able will break with the design of the base exception class itself, which is IIRC something people don't want in the SF project.
Besides there is no way this class can generate a useful/correct message based on the value of `$host` alone, the logic that figured out the `$host` value is incorrect and throws the exception knows what is wrong and can therefor set a meaningful message only.